gix-fs/tests/stack/mod.rs

```diff
@@ -35,97 +35,97 @@ fn p(s: &str) -> &Path {
     );
-    let bs_absolute = p("\\absolute");
+    let bs_looks_absolute = p(r"\absolute");
     assert!(
-        absolute.is_relative(),
+        bs_looks_absolute.is_relative(),
         "on Windows, strange single-backslash paths are relative (and relative to the current drive)"
     );
     assert_eq!(
-        p("relative").join(absolute),
-        absolute,
-        "relative + absolute = absolute - however, they kind of act like they are absolute in conjunction with relative base paths"
+        p("relative").join(looks_absolute),
+        looks_absolute,
+        "relative + unix-absolute = unix-absolute - the relative path without a drive is replaced"
     );
     assert_eq!(
-        p("relative").join(bs_absolute),
-        bs_absolute,
-        "relative + absolute = absolute - backslashes aren't special here, and it just acts like it's absolute"
+        p("relative").join(bs_looks_absolute),
+        bs_looks_absolute,
+        "relative + unix-absolute = unix-absolute - the relative path without a drive is replaced - backslashes aren't special here"
     );
 
     assert_eq!(
         p("c:").join("relative"),
         p("c:relative"),
-        "drive + relative = strange joined result with missing backslash, but it's a valid path that works just like `c:\relative`"
+        "drive + relative = relative to the drive-specific current directory"
     );
     assert_eq!(
-        p("c:\\").join("relative"),
-        p("c:\\relative"),
+        p(r"c:\").join("relative"),
+        p(r"c:\relative"),
         "absolute + relative = joined result"
     );
 
     assert_eq!(
-        p("\\\\?\\base").join(absolute),
-        p("\\\\?\\base\\absolute"),
+        p(r"\\?\base").join(looks_absolute),
+        p(r"\\?\base\absolute"),
         "absolute1 + unix-absolute2 = joined result with backslash"
     );
     assert_eq!(
-        p("\\\\.\\base").join(absolute),
-        p("\\\\.\\base\\absolute"),
+        p(r"\\.\base").join(looks_absolute),
+        p(r"\\.\base\absolute"),
         "absolute1 + absolute2 = joined result with backslash (device namespace)"
     );
     assert_eq!(
-        p("\\\\?\\base").join(bs_absolute),
-        p("\\\\?\\base\\absolute"),
+        p(r"\\?\base").join(bs_looks_absolute),
+        p(r"\\?\base\absolute"),
         "absolute1 + absolute2 = joined result"
     );
     assert_eq!(
-        p("\\\\.\\base").join(bs_absolute),
-        p("\\\\.\\base\\absolute"),
+        p(r"\\.\base").join(bs_looks_absolute),
+        p(r"\\.\base\absolute"),
         "absolute1 + absolute2 = joined result (device namespace)"
     );
 
     assert_eq!(p("/").join("C:"), p("C:"), "unix-absolute + win-drive = win-drive");
     assert_eq!(
         p("d:/").join("C:"),
         p("C:"),
-        "d-drive + c-drive = c-drive - interesting, as C: is supposed to be relative"
+        "d-drive + c-drive-relative = c-drive-relative - C: is relative but not on D:"
```

Comment:
This change relates to https://github.com/Byron/gitoxide/pull/1374#discussion_r1721055278.